### PR TITLE
Fix errors after protobuf 4.22 dump

### PR DESCRIPTION
**Pull request by denyszhak**
Fix errors after protobuf 4.22 dump
Closes https://github.com/python/typeshed/pull/9745

### CODE REVIEW

**Comment by denyszhak** [comment]
@AlexWaygood should I make changes to https://github.com/python/typeshed/pull/9745?

**Comment by hauntsaninja** [comment]
You should include the change to METADATA.toml that #9745 makes in this PR as well :-)

File: stubs/protobuf/google/protobuf/descriptor_pool.pyi [hunk 1]
```diff
@@ -2,7 +2,9 @@ from _typeshed import Incomplete
 
 class DescriptorPool:
     def __new__(cls, descriptor_db: Incomplete | None = ...): ...
```
Comment (AlexWaygood) [inline stubs/protobuf/google/protobuf/descriptor_pool.pyi:4]: This should fix the CI errors

```suggestion
    def __new__(cls, descriptor_db: Incomplete | None = ...): ...  # pyright: ignore[reportInconsistentConstructor]
```

File: stubs/protobuf/google/protobuf/descriptor_pool.pyi [hunk 2]
```diff
@@ -2,7 +2,9 @@ from _typeshed import Incomplete
...
-    def __init__(self, descriptor_db: Incomplete | None = ...) -> None: ...
+    def __init__(
```
Comment (AlexWaygood) [inline stubs/protobuf/google/protobuf/descriptor_pool.pyi:5]: This should fix the CI errors

```suggestion
    def __init__(  # pyright: ignore[reportInconsistentConstructor]
```

Comment (denyszhak) [reply stubs/protobuf/google/protobuf/descriptor_pool.pyi]: Thanks!

**Review by AlexWaygood — APPROVED** [review]
Thank you, looks good!

**Pull request merged**